docs/reference/function-configuration/stream-response.md

```diff
@@ -0,0 +1,115 @@
+# Streaming HTTP Responses
+
+Nuclio supports streaming HTTP responses for functions written in **Go** and **Python** runtimes. This feature is available **only for HTTP triggers** and is not supported for other trigger types.
+
+Streaming allows a function to send data to the client in chunks as it becomes available, rather than waiting for the entire response to be ready.
+
+## Supported Runtimes
+
+- Go
+- Python
+
+## Supported Triggers
+
+- HTTP
+
+---
+
+## Usage Examples
+
+### Go Runtime
+
+To stream a response in Go, use the `nuclio.NewResponseStream` object.
+
+Data can be sent in two ways:
+- `SendChunk`: Write a chunk of data (`[]byte`) directly to the stream.
+- `StreamFrom`: Stream data from any `io.Reader` (useful for external sources).
+
+**Always** use `defer responseStream.StopStreaming()` to properly close the stream.
+
+By default, `NewResponseStream` uses an `io.Pipe` internally to connect the function's output to the HTTP response.
+For advanced scenarios, a custom reader/writer can be provided using `NewCustomResponseStream`:
+
+```go
+responseStream := nuclio.NewCustomResponseStream(
+    "text/plain",
+    headers,
+    200,
+    customReader, // io.ReadCloser
+    customWriter, // io.Writer
+)
+```
+
+Example:
+
+```go
+package main
+
+import (
+    "fmt"
+    "strings"
+
+   "github.com/nuclio/nuclio-sdk-go"
+)
+
+func Streamer(context *nuclio.Context, event nuclio.Event) (interface{}, error) {
+    streamType := strings.ToLower(string(event.GetBody()))
+    responseStream := nuclio.NewResponseStream(
+        "text/plain",
+        map[string]interface{}{
+            "Content-Type": "text/plain",
+        },
+        200,
+    )
+
+    startStreamErr := make(chan error)
+
+    go func() {
+        defer responseStream.StopStreaming()
+        startStreamErr <- nil
+        for ch := 'A'; ch <= 'Z'; ch++ {
+            chunk := fmt.Sprintf("%c", ch)
+            responseStream.SendChunk([]byte(chunk))
+        }
+    }()
+    err := <-startStreamErr
+    return responseStream, err
+}
```

Comment:
While the code is pretty self-explantory, since it is an example I would still add a few comments - "initialize the stream response object", "stream data chunk by chunk", etc